homework/assignment_5/assignment_5.py

```diff
@@ -0,0 +1,458 @@
+
+    model_list = [
+        "Logistic Regression",
+        "SVM",
+        "KNN",
+        "Decision Tree",
+        "Random Forest",
+        "Naive Bayes",
+    ]
+    acc_list = []
+    auc_list = []
+    cm_list = []
+
+    for model in model_pipeline:
+        model.fit(X_train, y_train)
+        y_pred = model.predict(X_test)
+        acc_list.append(metrics.accuracy_score(y_test, y_pred))
+        fpr, tpr, _thresholds = metrics.roc_curve(y_test, y_pred)
+        auc_list.append(round(metrics.auc(fpr, tpr), 2))
+        cm_list.append(confusion_matrix(y_test, y_pred))
+
+    # comparing models
+    results_df = pd.DataFrame(
+        {
+            "Model": model_list,
+            "Accuracy": acc_list,
+            "AUC": auc_list,
+        }
+    )
+
+    """
+    Best model is Random Forest based on accuracy of prediction. Most likely over-fit due to the accuracy being 1.
+    Second best is Decision Tree with .99 accuracy followed by KNN at .98. Both probably over-fit...
+    Going off non over-fit models logistic regression actually performs the best at .78
+    Curious as to why RF and DT could be so overfit? I did not think that I had included predictors that 'cheated'
+    """
+
+    ##################################################
+    # TABLE TO HTML
+    ##################################################
+    # formatting for html out
+    html_string = """
+    <html>
+      <head><title>Assignment 5 Output</title></head>
+      <link rel="stylesheet" type="text/css" href="my_style.css"/>
+      <body>
+        {table}
+      </body>
+    </html>.
+    """
+    # sorting values
+    brute_force = brute_force.sort_values(
+        by=["Weighted Difference of Mean Response"], ascending=False
+    )
+    results_df = results_df.sort_values(by=["Accuracy"], ascending=False)
+
+    # creating clickable links
+    output_table = (
+        output_table.style.set_properties(**{"text-align": "center"})
+        .format({"Linear Regression Plot": make_clickable})
+        .hide_index()
+    )
+    brute_force = (
+        brute_force.style.set_properties(**{"text-align": "center"})
+        .format(
+            {
+                "Predictor 1 Bin Plot": make_clickable,
+                "Predictor 2 Bin Plot": make_clickable,
+            }
+        )
+        .hide_index()
+    )
+
+    # OUTPUT HTML FILE
+    with open("assignment_5.html", "w") as f:
+        f.write(
+            html_string.format(
+                table=output_table.to_html(
+                    justify="center", col_space=10, classes="mystyle"
+                )
+            )
+            + "\n\n"
+            + heat.to_html()
+            + "\n\n"
+            + html_string.format(
+                table=brute_force.to_html(justify="center", classes="mystyle")
+            )
+            + "\n\n"
+            + html_string.format(
+                table=results_df.to_html(justify="center", classes="mystyle")
+            )
+        )
```

Comment:
File runs, but it doesn't automatically for me. And I noticed you have some stats on features like home team vs away team, but you do have good plots that all have clickable links.